x/merkledb/db.go

```diff
@@ -422,6 +422,37 @@ func (db *merkleDB) Close() error {
 	return db.baseDB.Put(cleanShutdownKey, hadCleanShutdown)
 }
 
+func (db *merkleDB) PrefetchKeys(keys [][]byte) error {
+	db.commitLock.RLock()
+	defer db.commitLock.RUnlock()
+
+	if db.closed {
+		return database.ErrClosed
+	}
+
+	tempView, err := newTrieView(db, db, ViewChanges{})
+	if err != nil {
+		return err
+	}
+	for _, key := range keys {
+		pathToKey, err := tempView.getPathTo(db.newPath(key))
+		if err != nil {
+			return err
+		}
+		for _, n := range pathToKey {
```

Comment:
This will end up potentially adding the same node multiple times, but doesn't seem worth doing the deduplication since this is a simple write into a cache.